tests-beta/android/MASVS-CODE/MASTG-TEST-0222.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Position Independent Code (PIC) not enabled
+platform: android
+id: MASTG-TEST-0222
+type: [static]
+weakness: MASWE-0116
+---
+
+## Overview
+
+This test case checks if the shared libraries of the app are compiled without Position Independent Code (PIC) or [PIE (Position Independent Executable)](https://mas.owasp.org/MASTG/0x04h-Testing-Code-Quality/#position-independent-code), a common mitigation technique against memory corruption attacks.
+
+In general all binaries should be tested, which includes both the main app executable as well as all libraries/dependencies. However, Since Android 7.0 (API level 24), [PIC compilation is enabled by default](https://source.android.com/devices/tech/dalvik/configure) for the main executables. Therefore, on Android we will [focus on native libraries](../../../Document/0x05i-Testing-Code-Quality-and-Build-Settings/#binary-protection-mechanisms).
```

Comment:
```suggestion
In general all [native libraries](../../../Document/0x05i-Testing-Code-Quality-and-Build-Settings/#binary-protection-mechanisms) contained in the application should be tested.
```
I don't think this is applicable for the dex files of an Android app, or at least the developer won't have any control over it.